src/core/KeyDetector.ts

```diff
@@ -146,9 +150,23 @@ export class KeyDetector {
 
     return {
       type,
-      keys,
+      keys: Config.enableKeyPrefix ? detectedKeys(keys, document.getText()) : keys,
       locale,
       namespace,
     }
   }
 }
+
+function detectedKey(key: string, documentText?: string): string {
+  if (!documentText) return key
+  const mathces = documentText.match(/useTranslation\(.+,\s*{\s*keyPrefix:\s*["'`](?<keyPath>.+)["'`]\s*}\s*\)/)
+
+  const keyPath = mathces?.groups?.keyPath
+  if (!keyPath) return key
+
+  return `${keyPath}.${key}`
```

Comment:
I guess the key separator should be dynamic, as it can be configured in the i18nally configuration